pkg/datapath/linux/node.go

```diff
@@ -1148,7 +1148,7 @@ func (n *linuxNodeHandler) NodeValidateImplementation(nodeToValidate nodeTypes.N
 // NodeDeviceNameWithDefaultRoute returns the node's device name which
 // handles the default route in the current namespace
 func NodeDeviceNameWithDefaultRoute() (string, error) {
-	link, err := route.NodeDeviceWithDefaultRoute()
+	link, err := route.NodeDeviceWithDefaultRoute(option.Config.EnableIPv4, option.Config.EnableIPv6)
```

Comment:
Probably it would make sense to define a structure on each package that would specify its config? I'm thinking that if we add new options in the future that those won't make the function arguments increase as well.

Reply:
Agree in general and I will also follow up with more PRs splitting config per package.

In this specific case I think we would still need to translate from `pkg/option.Config` to the to-be-introduced config for `pkg/datapath/linux/route` (since we cannot just move `EnableIPv{4,6}` to this new config, as it's used in a lot of other places). And we would make sure to not forget setting this config this at the right place. IMO, this would currently make things more complicated. And also, having exported functions depend on package-global variables IMO makes it harder to reason about these functions and test them.

Maybe we can address this once the need arises, i.e. once we would need to add more function arguments?